File: app/src/main/java/com/base/BaseFragment.java
package com.base;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.kingz.GreekLifeCycle;
import com.zeke.ktx.App;

import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * author：KingZ
 * date：2019/7/26
 * description：基类的Fragment
 */
public abstract class BaseFragment extends Fragment {
    private static final String TAG = BaseFragment.class.getSimpleName();
    private Unbinder unbinder;
    protected Activity mActivity;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getLifecycle().addObserver(new GreekLifeCycle(TAG));
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    protected void bindButterKnife(View view){
        unbinder = ButterKnife.bind(this,view);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        mActivity = getActivity();
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mActivity = null;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        //解绑ButterKnife
        if(unbinder != null){
            unbinder.unbind();
        }
    }

    @Override
    public Context getContext() {
        return mActivity == null ? App.Companion.getInstance().getApplicationContext(): mActivity.getBaseContext();
    }

    public boolean onBackPressed() {
        return false;
    }

    public void showPlayBufferTips() {}
    public void dismissPlayBufferTips() {}
}

File: app/src/main/kotlin/com/zeke/ktx/base/BaseFragment.kt
package com.zeke.ktx.base

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.zeke.ktx.App
import com.zeke.ktx.AppLifeCycle
import com.zeke.ktx.view.LoadStatusView


/**
 * description: 基类的Fragment
 */
abstract class BaseFragment : Fragment() {
    companion object {
        val TAG = BaseFragment::class.java.simpleName
    }
    private var mActivity: Activity? = null
    protected lateinit var rootView: View
    // 加载状态View
    protected var loadStatusView: LoadStatusView? = null

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        mActivity = activity
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        lifecycle.addObserver(AppLifeCycle(TAG))
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        rootView = inflater.inflate(getLayoutId(), null)
        onCreateViewReady()
        return rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
        initData()
        setListener()
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
    }

    override fun onDetach() {
        super.onDetach()
        mActivity = null
    }

    override fun getContext(): Context? {
        return if (mActivity == null) App.instance else mActivity
    }

    fun onBackPressed(): Boolean {
        return true
    }

    /** 设置布局id*/
    abstract fun getLayoutId(): Int

    /** 进行初始化操作，在onCreateView中调用*/
    open fun onCreateViewReady() {}

    /** 初始化视图*/
    abstract fun initView()

    /** 初始化数据*/
    abstract fun initData()

    /** 设置监听*/
    open fun setListener() {}

}
